python/text-to-speech/test_main.py

```diff
@@ -0,0 +1,309 @@
+"""Unittests for Text To Speech Function."""
+# Standard library
+import base64
+import unittest
+import pathlib
+from unittest.mock import patch
+from unittest.mock import MagicMock
+import io
+
+# Third party
+import requests
+from parameterized import parameterized
+from google.cloud import texttospeech
+import boto3
+import botocore.response
+
+# Local imports
+import main
+
+# Path to Google encoded result (str).
+RESULT_GOOGLE = (
+    pathlib.Path("results/google.txt").
+    read_text(encoding="utf-8")
+)
+
+# Path to Azure encoded result (str).
+RESULT_AZURE = (
+    pathlib.Path("results/azure.txt").
+    read_text(encoding="utf-8")
+)
+# Path to Aws encoded result (str).
+RESULT_AWS = (
+    pathlib.Path("results/aws.txt").
+    read_text(encoding="utf-8")
+)
+
+# Content of Google result (decoded bytes).
+DECODED_RESULT_GOOGLE = base64.b64decode(RESULT_GOOGLE)
+# Content of Azure result (decoded bytes).
+DECODED_RESULT_AZURE = base64.b64decode(RESULT_AZURE)
+# Content of AWS result (decoded bytes).
+DECODED_RESULT_AWS = base64.b64decode(RESULT_AWS)
+
+
+class MyRequest:
+    """Class for defining My Request structure."""
+    def __init__(self, data: dict) -> None:
+        self.payload = data.get("payload", {})
+        self.variables = data.get("variables", {})
+
+
+class GoogleTest(unittest.TestCase):
+    """Google API Test Cases."""
+    def get_google_instance(self, key: str, project_id: str) -> main.Google:
+        """Set Google instance with request."""
+        req = MyRequest({
+            "payload": {
+                "provider": "google",
+                "text": "hi",
+                "language": "en-US",
+            },
+            "variables": {
+                "API_KEY": key,
+                "PROJECT_ID": project_id,
+            }
+        })
+        return main.Google(req)
+
+    @parameterized.expand([
+        # Missing API_KEY.
+        (None, "123"),
+        # Missing PROJECT_ID.
+        ("123", None),
+        # Missing Both.
+        (None, None),
+    ])
+    def test_validate_request(self, key: str, project_id: str) -> None:
+        """Test 'validate_request' method when fields missing or invalid."""
+        self.assertRaises(
+            ValueError,
+            self.get_google_instance,
+            key,
+            project_id,
+        )
+
```

Comment:
Here's another way you could write this:

```suggestion
        with self.assertRaises(ValueError):
            self.get_google_instance(key, project_id,)

```